core/contracts/financial-templates/implementation/Liquidatable.sol

```diff
@@ -287,8 +298,8 @@ contract Liquidatable is PricelessPositionManager {
         // Enqueue a request with the DVM.
         _requestOraclePrice(disputedLiquidation.liquidationTime);
 
-        // Pay a final fee
-        _payFinalFees(msg.sender);
+        // Pay a final fee.
+        _payFinalFees(msg.sender, disputedLiquidation.finalFee);
```

Comment:
Great point prasad brings up, also makes me wonder if we should consider having the flexibility to choose different final bond amounts for the liquidator and the disputer.

Depending on the amount of excess collateral, the disputer/liquidator rewards, and the dispute bond, there might be "more at stake" for one party

Reply:
@ptare you're 100% correct. I thought about it and couldn't come up with a way to pay the final fee at time of dispute and guarantee that the disputer would be fully refunded or even refunded to the point of there still being an incentive to dispute.

Allowing some free liquidation opportunity *before* a fee change might be acceptable if we didn't have to impose final fee changes through the voting system, giving everyone warning that it's coming for days beforehand.

If y'all have any ideas here (they definitely don't have to be perfect or elegant) where we don't have to lock in the final fee beforehand, I'm all ears.

Reply:
> Depending on the amount of excess collateral, the disputer/liquidator rewards, and the dispute bond, there might be "more at stake" for one party

My intuition is that this isn't a huge problem because we just need them to be positively incentivized in all cases, not have the same incentive, right?

Reply:
While this argument makes sense around incentives, if the final fee has increased from when the fee was calculated in the `createLiquidation` function then it is possible to under pay on the final fee relative to what is expected based off the current fee rate. Likewise if the fee has dropped then you over pay on the final fee. 

This doesn't really matter in the grand scheme of things because you are effectively locking in the fee that will be paid at liquidation time, rather than at dispute time. By doing this you ensure that both pay the same bond amount, calculated at liquidation time irrespective if this has deviated from the "true" fee amount.

My question is what are the implications for the store collecting the required fees in the case the fee rate drastically changes between `liquidate` (`finalFee` getting "locked in") and `dispute`? From an implementation perspective I dont think it matters as the `payOracleFeesErc20` in `FeePayer` pays the total `allowance` from the caller but could this result in incorrect bookkeeping between what is expected to be paid and what is actually paid?

Reply:
As we discussed IRL, this is definitely a tradeoff. We don't really do any bookkeeping in the store, so nothing would *break*, but it definitely doesn't match the precise expectations of the DVM.